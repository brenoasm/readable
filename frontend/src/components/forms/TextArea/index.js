import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import FormErrorMessage from '../FormErrorMessage';

const propTypes = {
  title: PropTypes.string,
  name: PropTypes.string,
  id: PropTypes.string,
  className: PropTypes.string,
  handleChange: PropTypes.func,
  errors: PropTypes.arrayOf(PropTypes.string)
};

const defaultProps = {
  title: '',
  name: '',
  id: '',
  className: '',
  handleChange: () => {},
  errors: []
};

const StyledTextArea = styled.div`
  margin: 5px;

  & > label {
    margin-right: 5px;
  }

  & > textarea {
    border: 2px solid rgb(238, 238, 238);
  }
`;

const TextArea = ({
  title,
  name,
  id,
  className,
  value,
  handleChange,
  errors
}) => {
  return (
    <StyledTextArea>
      <label htmlFor={name}>{title}</label>
      <textarea
        name={name}
        className={className}
        id={id}
        onChange={(event) => handleChange(event)}
        value={value}
      />
      <FormErrorMessage errors={errors} />
    </StyledTextArea>
  );
};

TextArea.propTypes = propTypes;
TextArea.defaultProps = defaultProps;

export default TextArea;
