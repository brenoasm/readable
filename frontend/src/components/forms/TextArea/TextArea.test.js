/* eslint-disable */

import React from 'react';

import TextArea from './';

describe('TextArea', () => {
  it('matches the snapshot', () => {
    const initialState = {
      name: 'author',
      title: 'author',
      value: 'author',
      id: 'author',
      handleChange: jest.fn(),
      errors: []
    };

    const wrapper = shallow(<TextArea {...initialState}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('shows a title when the property is passed', () => {
    const initialState = {
      title: 'author',
    };

    const wrapper = shallow(<TextArea {...initialState}/>);

    expect(wrapper.find('label').text()).toBe(initialState.title);
  });

  it('should have one FormErrorMessage', () => {
    const initialState = {
      errors: [
        'Erro ao inserir valor'
      ]
    };

    const wrapper = shallow(<TextArea {...initialState}/>);

    expect(wrapper.find('FormErrorMessage')).toHaveLength(1);
  });

  it('trigger the handleChange event when the value changes', () => {
    const initialState = {
      handleChange: jest.fn(),
    };

    const wrapper = shallow(<TextArea {...initialState}/>);

    wrapper.find('textarea').simulate('change', { target: { value: 'a' }});

    expect(initialState.handleChange).toHaveBeenCalled();
  });
});
