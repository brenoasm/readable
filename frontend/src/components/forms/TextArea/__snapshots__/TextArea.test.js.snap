// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextArea
    className=""
    errors={Array []}
    handleChange={[Function]}
    id="author"
    name="author"
    title="author"
    value="author"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label
          htmlFor="author"
>
          author
</label>,
        <textarea
          className=""
          id="author"
          name="author"
          onChange={[Function]}
          value="author"
/>,
        <FormErrorMessage
          errors={Array []}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "author",
          "htmlFor": "author",
        },
        "ref": null,
        "rendered": "author",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "",
          "id": "author",
          "name": "author",
          "onChange": [Function],
          "value": "author",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "errors": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label
            htmlFor="author"
>
            author
</label>,
          <textarea
            className=""
            id="author"
            name="author"
            onChange={[Function]}
            value="author"
/>,
          <FormErrorMessage
            errors={Array []}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "author",
            "htmlFor": "author",
          },
          "ref": null,
          "rendered": "author",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "",
            "id": "author",
            "name": "author",
            "onChange": [Function],
            "value": "author",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "errors": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
